.github/workflows/cd.yml

```
@@ -1,161 +1,160 @@
-  staging:
+  build:
+    if: github.ref == 'refs/heads/main'
     runs-on: ubuntu-latest
-    if: ${{ github.event.workflow_run.conclusion == 'success' }}
     environment:
       name: staging
+      # Staging domain (subdomain or free domain)
+      # Options: staging.yourdomain.com, pulseplate-staging.duckdns.org, or free Freenom domain
+      # Set STAGING_DOMAIN secret in GitHub Environment (e.g., "pulseplate-staging.duckdns.org")
+      # If secret not set, defaults to placeholder URL
+      url: ${{ secrets.STAGING_DOMAIN != '' && format('https://{0}', secrets.STAGING_DOMAIN) || 'https://staging.pulseplate.app' }}
```

actionlint (1.7.8)

context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)